Fix session join requirements documentation

xacrimon opened the pull request:
Followup, please see context in #14149.

ptgott commented on docs/pages/server-access/guides/tsh.mdx:
```diff
@@ -627,10 +627,23 @@ other user can join you through their terminal by typing:
+The Teleport user joining the session must have a `join_sessions` policy set up that allows
+them to join sessions created by the other user. If `joe` has a role named `access`, the other
+user needs have a role with a policy as shown below.
+More detailed information about `join_sessions` policies can be found in the [role reference](../../access-controls/reference.mdx) and [moderated sessions guide](../../access-controls/guides/moderated-sessions.mdx).
```
```suggestion
You can find more detailed information about `join_sessions` policies in the [role reference](../../access-controls/reference.mdx) and [Moderated Sessions guide](../../access-controls/guides/moderated-sessions.mdx).
```

ptgott replied:
I also think we should move this sentence into a separate paragraph below the code snippet, so we go directly from "as shown below" to the snippet.

ptgott commented on docs/pages/server-access/guides/tsh.mdx:
```diff
@@ -627,10 +627,23 @@ other user can join you through their terminal by typing:
-will be able to join.
+The Teleport user joining the session must have a `join_sessions` policy set up that allows
+them to join sessions created by the other user. If `joe` has a role named `access`, the other
+user needs have a role with a policy as shown below.
```
```suggestion
user needs have a role with a policy as shown below:
```

ptgott reviewed (commented):
Added some comments. At the same time, I'm not really sure if it's appropriate for this guide to include admin instructions related to setting up Teleport roles. If this guide is only for end users, the user running `tsh` will likely not have access to change their own roles. Is there another way for an end user to determine whether their cluster is set up to allow session joining without `tctl`/admin access?

zmb3 reviewed (approved):
thanks for this!

xacrimon commented:
@ptgott I've addressed your comments. That said I don't really know of a better place for this. I think this guide is followed by a lot of people wanting to try the feature out, cluster admins and others.

ptgott commented:
@xacrimon That's right—unless there's any other recourse for an end user without privileges to edit/list roles, I suppose the advice for end users would be "Tell your admin about this guide"

xacrimon committed:
adjust for new docs organisation

xacrimon commented:
@zmb3 @ptgott In accordance with that plan, I've revised this and added an admonition for users lacking access referring them and their cluster administrator to the moderated sessions guide. I also added a quicklink in the introductory paragraph of the moderated sessions guide to make it easier to find the section on join policies.

xacrimon committed:
Merge branch 'master' into joel/join-req-doc

ptgott commented on docs/pages/access-controls/guides/moderated-sessions.mdx:
```diff
@@ -12,6 +12,9 @@ other users to be present in a Server or Kubernetes Access session. Depending on
 the requirements, these users can observe the session in real time, participate
 in the session, and terminate the session at will.
 
+In addition, Teleport administrators can define rules that allow users to join eachothers
```
```suggestion
In addition, Teleport administrators can [define rules](#join_sessions) that allow users to join each other's
```

ptgott commented on docs/pages/access-controls/guides/moderated-sessions.mdx:
```diff
@@ -12,6 +12,9 @@ other users to be present in a Server or Kubernetes Access session. Depending on
 in the session, and terminate the session at will.
 
+In addition, Teleport administrators can define rules that allow users to join eachothers
+sessions from `tsh` and the Web UI. More on this [here](#join_sessions).
```
```suggestion
sessions from `tsh` and the Web UI.
```

xacrimon committed:
Update docs/pages/access-controls/guides/moderated-sessions.mdx

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

xacrimon committed:
Update docs/pages/access-controls/guides/moderated-sessions.mdx

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

xacrimon committed:
Merge branch 'master' into joel/join-req-doc
